### PR TITLE
VACMS-9956: add new section field to lovell menu items

### LINKED ISSUE #9956: CMS Lovell menu items should be identified as Lovell, Tricare, or both
~~When an an item is saved for Lovell, any node that is assigned to section "Lovell Federal health care"
AND added to the Lovell Federal health care menu should have an additional menu entry added to the Lovell Tricare menu so that editors don't have to add it to both.~~
image

~~Weighting of menu items is going to be challenging.   Ideally it is probably best to mimic the weight and parenting of the menu item saved in the main Lovell menu.   There are going to be challenges to this if the parent is not one that exists in both menus.  This should be rare and may just require a hand correction by an admin when it occurs, or we look up the parent name in the one menu and try to find the equivalent in the second menu.~~

After some discussion, @swirtSJW and I have come up with a new solution. Menu items are fieldable in Drupal 9 just like nodes, blocks and other entities. As such, we are proposing we add a new field to the Lovell Federal health care menu: /admin/structure/menu/manage/lovell-federal-health-care

This field should be called Menu Section, should be a list field and should allow for 3 values:
- main
- tricare
- both

When a node is saved, and a corresponding menu item is also saved, we check the section for the node the menu item points to. If that section is Lovell - VA the menu section for the menu item will be set to va. If that section is Lovell - Tricare the menu section will be set to tricare. If that section is Lovell Federal health care then the menu section will be set to both.

Additional work will be needed on the front end to read these values in the appropriate graphQL queries to show the appropriate menu items:

- The Lovell Tricare page should show menu items for tricare and both while the Lovell VA page should show menu items for va and both. 

## AC's
- [ ] Code lives in new vamc_lovell module, menu presave function
- [ ] When a menu item is saved check what menu it lives in.
- [ ] If the menu item is being saved in the Lovell Federal health care menu see if the menu item is a link
- [ ] If it is a link and it points to a drupal entity, load that entity and check the value of field_administration
- [ ] Set the field_menu_section value for the menu item based on the value in field_administration on the node.
- [ ] ~~When an item is saved for Lovell, any node that is assigned to section "Lovell Federal health care"
AND added to the Lovell Federal health care menu should have an additional menu entry added to the Lovell tricare menu so that editors don't have to add it to both.~~
- [ ] ~~Menu items added to both, end up in the right spot in the Tricare menu.~~
- [ ]  ~~When parents can't be found, people are notified that placement in Lovell Tricare failed and to contact cms help desk ???~~

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #9956 

## Testing done

Manual testing

## Screenshots

Screen Shot 2022-08-10 at 9 29 32 AM

Screen Shot 2022-08-10 at 9 30 00 AM

## QA steps

As an admin user

- [x]  Visit /admin/structure/menu/manage/lovell-federal-health-care in one tab to get a list of links in the Lovell menu
- [x] Click manage fields and verify field_menu_section has been added to the menu
- [x] Click edit next to field_menu_section and then click field settings to verify the three available values
- [x] Go back to /admin/structure/menu/manage/lovell-federal-health-care and click edit next to the first link (opening in a new tab) - verify the existence of the Menu Section field at the bottom of the form (should be set to - None - initially)
- [x] Also click on the first link itself to open the node it points to in a new tab
- [x] Try editing and saving the node, changing the section from Lovell, Lovell - VA and Lovell - Tricare. After each save check the menu item edit page to verify the Menu Section has updated in kind.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [x] `⭐️ Facilities`
- [ ] `⭐️ User support`

### CODE REVIEW

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This is looking really good Jay.  Nice work.   I think there is just one hole that I see in the logic.

File: docroot/modules/custom/va_gov_lovell/src/EventSubscriber/LovellEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        if ($node->hasField('field_administration')) {
+          $section_id = $node->get('field_administration')->target_id;
+          $section_map = [
+            '1040' => 'va',
+            '1039' => 'tricare',
+            '347' => 'both',
+          ];
+
+          // Set section for this menu item.
+          $entity->set('field_menu_section', $section_map[$section_id]);
```
[the remark this answers is not in the document]
Comment (JayDarnell) [reply docroot/modules/custom/va_gov_lovell/src/EventSubscriber/LovellEventSubscriber.php:98]: Good catch. Thanks @swirtSJW

**Review by swirtSJW — COMMENTED** [review]
Looks great.  Nice work

**Review by swirtSJW — APPROVED** [review]
Tests out good
image

image


image

**Comment by swirtSJW** [comment]
image

**Pull request merged**